library/src/main/java/com/chuckerteam/chucker/internal/support/TeeSource.kt

```diff
@@ -26,9 +23,8 @@ internal class TeeSource(
 ) : Source {
     private val sideStream = Okio.buffer(Okio.sink(sideChannel))
     private var totalBytesRead = 0L
-    private var isReadLimitExceeded = false
-    private var isUpstreamExhausted = false
     private var isFailure = false
+    private var isClosed = false
```

Comment:
Do we really need those two? Is there a risk `.onClosed` or `.onFailure` gets called twice?

Reply:
`isClosed` flag should be here. Otherwise closing the source multiple times would trigger callback that many times as well and closing should generally be an idempotent operation.

`isFalure` can be removed. Personally I don't think it makes sense to report all failures and multiple can occur while writing to a file. If a single write fails (due to some file corruption on a low-end device or whatever) there is a high chance it will fail again. This way we can skip writing If a failure occurred and report only a first failure (either network IO or disk IO).

Reply:
Great thanks for clarifying